src/ui/units/dash/containers/Body/Body.tsx

```diff
@@ -180,18 +181,27 @@ const GROUPS_WEIGHT = {
 // Body is used as a core in different environments
 class Body extends React.PureComponent<BodyProps> {
     static getDerivedStateFromProps(props: BodyProps, state: DashBodyState) {
+        let newState: Partial<DashBodyState> = {};
+
         const {
             prevMeta: {entryId, tabId},
         } = state;
-
         // reset loaded before new tab/entry items are mounted
         if (props.entryId !== entryId || props.tabId !== tabId) {
             state.loadedItemsMap.clear();
+            newState = {
+                prevMeta: {tabId: props.tabId, entryId: props.entryId},
+                loaded: false,
+            };
+        }
 
-            return {prevMeta: {tabId: props.tabId, entryId: props.entryId}, loaded: false};
+        const newHash = props.location.hash;
+        if (newHash !== state.hash) {
+            newState.hash = newHash;
+            scrollToHash({hash: newHash.replace('#', ''), withDelay: props.tabId !== tabId});
```

Comment:
You do `hash.replace('#', '')` inside `scrollToHash`. It seems that it is not necessary to do this here?